Mathlib/Algebra/Homology/DerivedCategory/Ext/Basic.lean

```diff
@@ -172,6 +186,33 @@ lemma mk₀_comp_mk₀_assoc (f : X ⟶ Y) (g : Y ⟶ Z) {n : ℕ} (α : Ext Z T
   rw [← mk₀_comp_mk₀, comp_assoc]
   omega
 
+
+variable (X Y) in
+lemma mk₀_bijective : Function.Bijective (mk₀ (X := X) (Y := Y)) := by
+  letI := HasDerivedCategory.standard C
+  have h : (singleFunctor C 0).FullyFaithful := Functor.FullyFaithful.ofFullyFaithful _
+  let e : (X ⟶ Y) ≃ Ext X Y 0 :=
+    (h.homEquiv.trans (ShiftedHom.homEquiv _ (by simp))).trans homEquiv.symm
+  have he : e.toFun = mk₀ := by
+    ext f : 1
+    dsimp [e]
+    apply homEquiv.injective
+    apply (Equiv.apply_symm_apply _ _).trans
```

Comment:
This is a funny trick! Do you do everything backwards (see the `symm` on line 201 and the backwards `rw` on line 203) so you can use this trick? Why does `rw [Equiv.apply_symm_apply homEquiv]` not work?

Reply:
`erw [Equiv.apply_symm_apply]` would work, but not `rw`. This is because somewhere there is a the coercion of `0 : ℕ` to `0 : ℤ` (a definitional equality but not a syntactic equality) in `homEquiv {n : ℕ} : Ext.{w} X Y n ≃ ShiftedHom ((singleFunctor C 0).obj X) ((singleFunctor C 0).obj Y) (n : ℤ)` for the natural number `n := 0`.